src/main/java/org/hiero/sketch/spreadsheet/Histogram1D.java

```diff
@@ -0,0 +1,87 @@
+package org.hiero.sketch.spreadsheet;
+
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.hiero.sketch.table.api.IColumn;
+import org.hiero.sketch.table.api.IMembershipSet;
+import org.hiero.sketch.table.api.IRowIterator;
+import org.hiero.sketch.table.api.IStringConverter;
+
+public class Histogram1D implements IHistogram {
+    final private IColumn column;
+    final private IMembershipSet membershipSet;
+    final private double[] boundaries;
+    final private int[] buckets;
+    final private IStringConverter converter;
+    private int missingData;
+    private int outOfRange;
+
+    public Histogram1D(@NonNull final IColumn column, @NonNull final IMembershipSet membershipSet,
+                       @NonNull final IStringConverter converter, @NonNull final double[] boundaries) {
+        if (!isSorted(boundaries))
+            throw new IllegalArgumentException("Bucket array has to be sorted");
+        this.column = column;
+        this.membershipSet = membershipSet;
+        this.boundaries = boundaries;
+        this.buckets = new int[boundaries.length - 1];
+        this.converter = converter;
+        this.createHistogram();
+    }
+
+    /**
+     * Creates the histogram explicitly and in full.
+     */
+    private void createHistogram() {
+        final IRowIterator myIter = this.membershipSet.getIterator();
+        int currIndex = myIter.getNextRow();
+        while (currIndex >= 0) {
+            if (this.column.isMissing(currIndex))
+                this.missingData++;
+            else
+                placeInBucket(this.column.asDouble(currIndex, this.converter));
+            currIndex = myIter.getNextRow();
+        }
+    }
+
+    private void placeInBucket(final double entry) {
+        final int index = Histogram1D.indexOf(this.boundaries, entry);
+        if (index >= 0)
+            this.buckets[index]++;
+        else this.outOfRange++;
+    }
+
+    // Todo: Move to a base class at some point
+    private static boolean isSorted(final double[] a) {
+        for (int i = 0; i < (a.length - 1); i++) {
+            if (a[i] > a[i + 1]) {
```

Comment:
It's a bug that got fixed 'automatically', under the assumption that right inclusive is always false the array has to be strongly sorted. I'll document it.